instrumentation/github.com/gin-gonic/gin/otelgin/option.go

```diff
@@ -57,3 +59,19 @@ func WithTracerProvider(provider oteltrace.TracerProvider) Option {
 		}
 	})
 }
+
+// WithSpanStartOptions configures an additional set of
+// trace.SpanOptions, which are applied to each new span.
+func WithSpanStartOptions(opts ...oteltrace.SpanStartOption) Option {
+	return optionFunc(func(cfg *config) {
+		cfg.SpanStartOptions = append(cfg.SpanStartOptions, opts...)
```

Comment:
My understanding of "with" is to cover, rather than to append.

If it were to append, I might consider naming it: `AppendSpanStartOptions`.

> `WithSpanEndOptions`  is the same.